File: LensAccess.py
import LensCtrl
import DefVal as DV

fineNum = 10    #default focus fine tuning number

def MoveLens(motor):
    data = input("\nEnter the adderess you want to go ")
    addrData = int(data)
    if (motor == DV.ZOOM):
        LensCtrl.ZoomMove(addrData)
    elif (motor == DV.FOCUS):
        LensCtrl.FocusMove(addrData)
    elif (motor == DV.IRIS):
        LensCtrl.IrisMove(addrData)
    elif (motor == DV.OPT):
        LensCtrl.OptFilterMove(addrData)
    print("")


def UserIDRead():
    retval, userName = LensCtrl.UserAreaRead()
    print(userName)

def UserIDWrite():
    print("\nPlease fill in  the user area(32byte full)")
    print("12345678901234567890123456789012")
    userName = input("")
    if len(userName) > 32:
        print("Error! Only 32 characters allowed!")
    else:
        LensCtrl.UserAreaWrite(userName)
    print("")

def FineFocusMove():
    print("Fine focus move")
    print("Selection : Near -> n / Far -> f / End -> q, fine step = {} >>".format(fineNum), end="")
    dir = ""
    while True:
        key = input()
        if(key == ("n" or"N")):
            dir = "Near"
            print("Set Near. Press Enter key.", end="")
        elif(key == ("f" or "F")):
            dir = "Far"
            print("Set Far. Press Enter key.", end="")

        elif(key ==""):
            if(dir =="Near"):
                addrData = LensCtrl.focusCurrentAddr + fineNum
                LimFocusMove(addrData)
            elif(dir =="Far"):
                addrData = LensCtrl.focusCurrentAddr - fineNum
                LimFocusMove(addrData)

        elif(key ==("q" or "Q")):
            dir = "Quit"
            break

def LimFocusMove(addrData):

    if ((LensCtrl.focusMaxAddr >= addrData) & (addrData > LensCtrl.focusMinAddr)):
        retval = LensCtrl.FocusMove(addrData)
        if (retval == DV.RET_SUCCESS):
            print("Addr = {} >> ".format(LensCtrl.focusCurrentAddr), end="")
    else:
        print("Limit", end= "")

def FineNumSet():
    global fineNum
    num = input("Current fine number = {} \nFine move step number = ".format(fineNum))
    fineNum = int(num)

File: LensConnect_SampleProgram_Linux_C_x86_x64_1.0.0/LensConnect_Controller/C/LensAccess.c
#include"LensCtrl.h"
#include"LensConnect.h"
#include<unistd.h>
#include<termios.h>

void MoveLens(int motor) {
	int addrData = 0;
	printf("\nEnter the address you want to go ");
	scanf("%u", &addrData);
	switch (motor) {
	case ZOOM:
		ZoomMove((uint16_t)addrData);
		break;
	case FOCUS :
		FocusMove((uint16_t)addrData);
		break;
	case IRIS:
		IrisMove((uint16_t)addrData);
		break;
	case OPT:
		OptFilterMove((uint16_t)addrData);
		break;
	}
	printf("\n");
}
void UserIDRead() {
	char userName[33];
	UserAreaRead(userName);
	printf("%s\n", userName);

}
void UserIDWrite() {
	char userName[33];
	char dummy;
	printf("\nPlease fill in the user area(32byte full)\n");
	printf("12345678901234567890123456789012\n");
	scanf("%[\n]",&dummy);
	scanf("%[^\n]", userName);
	UserAreaWrite(userName);
	printf("\n");
}

void FineFocusMove(){
#define QUIT_CHAR 0x0d	// Enter key
struct termios CookedTermIos;
struct termios RawTermIos;
	tcgetattr(STDIN_FILENO, &CookedTermIos);
	RawTermIos = CookedTermIos;
	cfmakeraw(&RawTermIos);
	tcsetattr(STDIN_FILENO, 0, &RawTermIos);
	int c;
	uint16_t addrData;
	while ((c = getchar()) != QUIT_CHAR) {
		switch(c) {
		case 'z':
			addrData = focusCurrentAddr + fineNum;
			LimFocusMove(addrData);
			break;
		case 'x':
			addrData = focusCurrentAddr - fineNum;
			LimFocusMove(addrData);
			break;
		default :
			break;
		}
	}
	printf("\r\nEnd fine move focus mode\n");
	tcsetattr(STDIN_FILENO, 0, &CookedTermIos);
	putchar('\n');
}

void LimFocusMove(uint16_t addrData) {
	if ((focusMaxAddr >= addrData) & (addrData >= focusMinAddr)){
		int retval = FocusMove(addrData);
		if (retval == SUCCESS)
			printf("\rAddr= %d", focusCurrentAddr);
	} else {
		printf("\rLimit      ");
	}
}	

